### PR TITLE
Fix printing when show called on invalid variable or constraint

### PR DESCRIPTION
**Pull request by odow**
Closes https://github.com/jump-dev/JuMP.jl/issues/3762

 - [x] Test extensions: https://github.com/jump-dev/JuMP.jl/actions/runs/9312266581

### CODE REVIEW

File: test/test_print.jl [hunk 1]
```diff
@@ -1070,4 +1070,21 @@ function test_print_omit_vector()
     return
 end
 
+function test_invalid_references()
+    model = Model()
+    @variable(model, x[1:2])
+    @constraint(model, c[i=1:2], x[i] <= i)
```
Comment (odow) [inline test/test_print.jl:1076]:
```suggestion
    @constraint(model, c[i in 1:2], x[i] <= i)
```

**Review by blegat — APPROVED** [review]
We can ignore PolyJuMP and SumOfSquares failing, I'm doing a big refactoring

**Comment by odow** [comment]
So you're okay with just printing "InvalidXXX"?

**Comment by blegat** [comment]
Yes, the variable is invalid so using it in a model should throw but we could allow printing it and `InvalidXXX` seems reasonable.

**Pull request merged**